### PR TITLE
Blackjack deal

### PR DESCRIPTION
**Pull request by mackmc22**
getting the following error in the server: TypeError: view() missing 1 required positional argument: 'request'

I have googled and played around with it, but I have no idea what the issue is. Advice?

### CODE REVIEW

File: blackjack/api/views.py [hunk 1]
```diff
@@ -8,7 +9,15 @@
...
+    deck_cards = [2, 2, 2, 2, 3, 3, 3, 3, 4, 4, 4, 4, 5, 5, 5, 5, 6, 6, 6, 6, 7, 7, 7, 7, 8, 8, 8, 8, 9, 9, 9, 9,
+                  10, 10, 10, 10, 'J', 'J', 'J', 'J', 'Q', 'Q', 'Q', 'Q', 'K', 'K', 'K', 'K', 'A', 'A', 'A', 'A'
+                  ]
 
-@api_view(["GET"])
-def deal(request):
-    return JsonResponse(data={'key':'value'}, status=status.HTTP_200_OK)
\ No newline at end of file
+    def __init__(self):
+        random.shuffle(self.deck_cards)
```
Comment (de1ux) [inline blackjack/api/views.py:18]: 🎆 your python skills are on point 🎆

File: blackjack/api/views.py [hunk 2]
```diff
@@ -8,7 +9,15 @@
...
+
+    @api_view(["GET"])
+    def deal(self, request):
```
Comment (de1ux) [inline blackjack/api/views.py]: I wouldn't make this a method on the Game class. Drop this back to a regular function (no `self`), and instantiate an instance of Game before dealing a card.

File: blackjack/blackjack/urls.py [hunk 3]
```diff
@@ -22,6 +22,6 @@
 urlpatterns = [
     path('admin/', admin.site.urls),
 
-    path("api/game", views.deal, ),
+    path("api/game", views.Game.deal(), ),
```
Comment (de1ux) [inline blackjack/blackjack/urls.py]: this is good intuition (deal as a method of the game class), but see above

File: blackjack/api/views.py (hunk 1 above)
Comment (mackmc22) [reply blackjack/api/views.py:18]: full disclosure, I did steal this from our previous python code of the game so don't get too excited

File: blackjack/blackjack/urls.py (hunk 3 above)
Comment (mackmc22) [reply blackjack/blackjack/urls.py]: fixed!

File: blackjack/api/views.py (hunk 2 above)
Comment (mackmc22) [reply blackjack/api/views.py]: that worked! thank you! 

as a general rule, are endpoints not methods on a class even if it is a function of the class? does that make sense?

**Pull request merged**